include/dt-bindings/gpio/gpio.h

```diff
@@ -133,48 +177,128 @@
+ *
+ * @deprecated Replace with `GPIO_OPEN_SOURCE`
  */
-#define GPIO_DS_DISCONNECT_LOW (0x3 << GPIO_DS_LOW_POS)
+#define GPIO_DS_DISCONNECT_LOW (0x3 << GPIO_DS_LOW_SHIFT)
 
-/** @cond INTERNAL_HIDDEN */
-#define GPIO_DS_HIGH_POS 14
-#define GPIO_DS_HIGH_MASK (0x3 << GPIO_DS_HIGH_POS)
-/** @endcond */
+/** Legacy alias identifying high drive strength when GPIO output is high.
+ *
+ * @deprecated Replace with `GPIO_DS_HI_HIGH`
+ */
+#define GPIO_DS_ALT_HIGH        GPIO_DS_HI_HIGH
 
-/** Default drive strength when GPIO pin output is high.
+/** Legacy flag indicating pin is disconnected when GPIO pin output is high.
+ *
+ * @deprecated Replace with `GPIO_OPEN_DRAIN`
  */
-#define GPIO_DS_DFLT_HIGH (0x0 << GPIO_DS_HIGH_POS)
+#define GPIO_DS_DISCONNECT_HIGH (0x3 << GPIO_DS_HIGH_SHIFT)
 
-/** Alternative drive strength when GPIO pin output is high.
- * For hardware that does not support configurable drive strengths
- * use the default drive strength.
+/** Legacy flag indicating that interrupts should be enabled on the GPIO.
+ *
+ * @deprecated Replace with `GPIO_INT_ENABLE`
  */
-#define GPIO_DS_ALT_HIGH (0x1 << GPIO_DS_HIGH_POS)
+#define GPIO_INT                GPIO_INT_ENABLE
 
-/** Disconnect pin when GPIO pin output is high.
- * For hardware that does not support disconnect use the default
- * drive strength.
+/** Legacy flag setting indicating signal or interrupt active level.
+ *
+ * This flag was used both to indicate a signal's active level, and to
+ * indicate the level associated with an interrupt on a signal.  As
+ * active level is also relevant to output signals the two
+ * intepretations have been separated.  The legacy value supports
+ * testing for interrupt level as this is the most common use in
+ * existing code.
+ *
+ * @deprecated Replace with `GPIO_ACTIVE_LOW` or `GPIO_INT_LOW`
+ * depending on intent.
  */
-#define GPIO_DS_DISCONNECT_HIGH (0x3 << GPIO_DS_HIGH_POS)
-/** @} */
+#define GPIO_INT_ACTIVE_LOW     GPIO_INT_LOW
+
+/** Legacy flag setting indicating signal or interrupt active level.
+ *
+ * This flag was used both to indicate a signal's active level, and to
+ * indicate the level associated with an interrupt on a signal.  As
+ * active level is also relevant to output signals the two
+ * intepretations have been separated.  The legacy value supports
+ * testing for interrupt level as this is the most common use in
+ * existing code.
+ *
+ * @deprecated Replace with `GPIO_ACTIVE_HIGH` or `GPIO_INT_HIGH`
+ * depending on intent.
+ */
+#define GPIO_INT_ACTIVE_HIGH    GPIO_INT_HIGH
 
+/** Legacy flag indicating interrupt triggers on both rising and falling edge.
+ *
+ * @deprecated Replace with `GPIO_INT_EDGE_BOTH`.
+ */
+#define GPIO_INT_DOUBLE_EDGE    (1 << 17)
+
+/** Legacy flag indicating a desire to debounce the signal.
+ *
+ * @deprecated Replace with `GPIO_INPUT_DEBOUNCE`
+ */
+#define GPIO_INT_DEBOUNCE       GPIO_INPUT_DEBOUNCE
 
+/** @cond INTERNAL_HIDDEN */
+#define GPIO_POL_SHIFT		18
```

Comment:
If part of your motivation for rejecting #11880 was objection to defining offset macros for single-bit fields, you really ought to remove this one.

Reply:
All GPIO_POL_ flags were deprecated and are going to be removed in the future. `GPIO_POL_SHIFT` is required to define `GPIO_POL_MASK` which is used in the code base.

Reply:
So there is existing precedent in Zephyr for providing mask and shift defines for single-bit fields.  Good to know.